packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { ReactElement, useEffect } from 'react';
+import { Modal, ModalProps } from '../common/Modal';
+import { ModalClose } from '../common/ModalClose';
+import { ButtonIconPosition, ButtonVariant } from '../../buttons/common';
+import { Image } from '../../image/Image';
+import { bookmarkFolderSoonImage } from '../../../lib/image';
+import {
+  Typography,
+  TypographyColor,
+  TypographyTag,
+  TypographyType,
+} from '../../typography/Typography';
+import {
+  useActions,
+  usePlusSubscription,
+  useViewSize,
+  ViewSize,
+} from '../../../hooks';
+import { ActionType } from '../../../graphql/actions';
+import { Button } from '../../buttons/Button';
+import { ArrowIcon } from '../../icons';
+import { plusUrl } from '../../../lib/constants';
+import { LogEvent, TargetId } from '../../../lib/log';
+import Link from '../../utilities/Link';
+
+type BookmarkFolderSponsorModalProps = Omit<ModalProps, 'children'>;
+
+export const BookmarkFolderSponsorModal = (
+  props: BookmarkFolderSponsorModalProps,
+): ReactElement => {
+  const { onRequestClose } = props;
+  const { checkHasCompleted, isActionsFetched, completeAction } = useActions();
+  const { logSubscriptionEvent } = usePlusSubscription();
+  const isTablet = useViewSize(ViewSize.Tablet);
+
+  useEffect(() => {
+    if (
+      isActionsFetched &&
+      !checkHasCompleted(ActionType.CreateBookmarkFolder)
+    ) {
+      completeAction(ActionType.CreateBookmarkFolder);
+    }
+  }, [checkHasCompleted, completeAction, isActionsFetched]);
+
+  return (
+    <Modal
+      kind={Modal.Kind.FlexibleCenter}
+      size={Modal.Size.Small}
+      isDrawerOnMobile
+      {...props}
+    >
+      <ModalClose
+        className="right-4 top-4"
+        onClick={onRequestClose}
+        variant={ButtonVariant.Primary}
+      />
+      <Modal.Body className="!tablet:p-4 flex flex-col justify-center gap-4 tablet:items-center tablet:text-center">
+        <div className="relative overflow-hidden">
+          <Image
+            className="rounded-16"
+            src={bookmarkFolderSoonImage}
+            alt="Bookmark Folder coming soon"
+          />
+        </div>
+        <Typography
+          bold
+          color={TypographyColor.Primary}
+          type={isTablet ? TypographyType.Title1 : TypographyType.LargeTitle}
```

Comment:
This is also new, haven't seen it on a modal yet I don't think?

Reply:
Should I ask to design to change it? This is from figma

Reply:
Hmm yeah just verify if it's intended and we normally don't do this?

Reply:
Design want to keep those changes ✔️